docs/relationships.md

```diff
@@ -5,35 +5,113 @@ title: Relationships
 
 ## Database Relationships
 
-Graphback supports `one-to-one` and `one-to-many` relationships and provides out of the box support for the schema and resolvers accordingly.
+Graphback supports `one-to-one`, `one-to-many` and `many-to-one` relationships and provides out of the box support for the schema and resolvers accordingly.
 
-### One To One
+### OneToOne
+
+The `1:1` relation can be declared by:
 
-The `1:1` relation can be simply declared by:
 ```graphql
 type Profile {
   user: User!
 }
+
+type User {
+  profile: Profile!
+}
 ```
-which will create the relationship via a column in the `user` table using column `profileId`. You can customize the field which tracks the relationship using directives.
+
+which will create the relationship via a `userId` column in the `profile` table and a `profileId` column in the `user` table. You can customize the field which tracks the relationship using annotations:
+
 ```graphql
 type Profile {
-  user: User! @OneToOne(field: "yourCustomField")
+  profileUser: User!
+}
+
+type User {
+  """
+  @db.oneToOne: 'profileUser'
+  """
+  profile: Profile!
 }
 ```
 
-### One To Many
+This creates a `1:1` relationship between `profile.profileUserId` and `user.profileId`.
+
+### ManyToOne
+
+The `m:1` relation is declared like so:
 
-The `1:m` relation is declared with array syntax.
 ```graphql
-type Note {
-  comment: [Comment!]!
+type Comment {
+  note: Note!
 }
 ```
-This creates a column `noteId` in table `comment` to track the relationship. You can similarily customize the column name as follows
+
+This creates a `noteId` column in the `comment` table to track the relationship.
+
+You can also customise the field name:
+
 ```graphql
+type Comment {
+  customNote: Note
+}
+```
+
+This creates `customNoteId` in the `comment` table.
+
+You can track to custom field names in the relation table using the `@db.manyToOne` annotation.
+
+```gql
+type Comment {
+  """
+  @db.manyToOne: 'noteComments'
+  """
+  note: Note
+}
+```
+
+This tracks to `Note.noteComments`.
+
+### OneToMany
+
+The `1:m` relationship does not alter the database structure, but can be used to track relationships for querying and subscriptions.
```

Comment:
This is very confusing thing. Initially I read it as this is going to give me something. 
I will simply merge both together and explain how to make `OneToMany` relationship.
This form of relationship is more common. 

Generally if we accept the way that db migration works we need to come up with the docs that make clear how to do it. I see so many issues with people having only `  comments: [Comment]`.

Without proper validation, this can happen so maybe we need to revisit this